src/promise-pool-executor.ts

```diff
@@ -169,12 +169,12 @@ export class PromisePoolExecutor<T, R> {
    * @returns {Promise}
    */
   async process (): Promise<ReturnValue<T, R>> {
-    for (const item of this.items) {
+    for (const [i, item] of Object.entries(this.items)) {
```

Comment:
Object.entries is giving the key as a string. Let's use the Array.entries method (`this.items.entries()`) to provide the index as an integer.